src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -236,12 +236,21 @@ private IList<string> GetResourceNamesForCulture(CultureInfo culture)
             {
                 var names = new List<string>();
                 using (var cultureResourceStream = _resourceAssemblyWrapper.GetManifestResourceStream(resourceStreamName))
-                using (var resources = new ResourceReader(cultureResourceStream))
                 {
-                    foreach (DictionaryEntry entry in resources)
+                    if (cultureResourceStream != null)
                     {
-                        var resourceName = (string)entry.Key;
-                        names.Add(resourceName);
+                        using (var resources = new ResourceReader(cultureResourceStream))
+                        {
+                            foreach (DictionaryEntry entry in resources)
+                            {
+                                var resourceName = (string)entry.Key;
+                                names.Add(resourceName);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        throw new MissingManifestResourceException();
```

Comment:
Could this include the resource name you were looking for? Would help in diagnosing the issue.